selfdrive/car/docs_definitions.py

```diff
@@ -31,60 +31,117 @@ class Star(Enum):
   EMPTY = "empty"
 
 
-class Harness(Enum):
-  nidec = "Honda Nidec"
-  bosch_a = "Honda Bosch A"
-  bosch_b = "Honda Bosch B"
-  toyota = "Toyota"
-  subaru_a = "Subaru A"
-  subaru_b = "Subaru B"
-  fca = "FCA"
-  ram = "Ram"
-  vw = "VW"
-  j533 = "J533"
-  hyundai_a = "Hyundai A"
-  hyundai_b = "Hyundai B"
-  hyundai_c = "Hyundai C"
-  hyundai_d = "Hyundai D"
-  hyundai_e = "Hyundai E"
-  hyundai_f = "Hyundai F"
-  hyundai_g = "Hyundai G"
-  hyundai_h = "Hyundai H"
-  hyundai_i = "Hyundai I"
-  hyundai_j = "Hyundai J"
-  hyundai_k = "Hyundai K"
-  hyundai_l = "Hyundai L"
-  hyundai_m = "Hyundai M"
-  hyundai_n = "Hyundai N"
-  hyundai_o = "Hyundai O"
-  hyundai_p = "Hyundai P"
-  hyundai_q = "Hyundai Q"
-  custom = "Developer"
-  obd_ii = "OBD-II"
-  gm = "GM"
-  nissan_a = "Nissan A"
-  nissan_b = "Nissan B"
-  mazda = "Mazda"
-  ford_q3 = "Ford Q3"
-  ford_q4 = "Ford Q4"
-  none = "None"
-
-
-class HarnessPart(Enum):
-  harness_box = "harness box"
-  comma_power_v2 = "comma power v2"
-  rj45_cable = "RJ45 cable (7 ft)"
-  long_obdc_cable = "long OBD-C cable"
-  usbc_coupler = "USB-C coupler"
-
-
-DEFAULT_HARNESS_PARTS: List[HarnessPart] = [HarnessPart.harness_box, HarnessPart.comma_power_v2, HarnessPart.rj45_cable]
+class PartType(Enum):
+  connector = "Connector"
+  device = "Device"
+  cable = "Cable"
+  accessory = "Accessory"
+  mount = "Mount"
 
 
 @dataclass
-class HarnessKit:
-  connector: Harness = Harness.none
-  parts: List[HarnessPart] = field(default_factory=lambda: copy.copy(DEFAULT_HARNESS_PARTS))
+class AbstractDataClass:
+  name: str
```

Comment:
why both this and `Part`?

Reply:
Why no dataclass? Can you not make a base abstract dataclass?

Reply:
mypy does not support a class being both a dataclass and having abstractmethod ([see this issue](https://github.com/python/mypy/issues/5374)).

We could have done something like this to make it work:
```
@dataclass
class PartDataClass:
  name: str


class PartAbstractClass(ABC):
  @property
  @abstractmethod
  def type(self) -> PartType:
    raise NotImplementedError


class Part(PartAbstractClass, PartDataClass):
  pass


class Connector(Part):
  @property
   def type(self) -> PartType:
     return ....

```

but I prefered just removing the dataclass from `Part` and just adding an `__init__` which gives the same result but is cleaner imo
```
class Part(ABC):
  def __init__(self, name: str):
    self.name = name

  @property
  @abstractmethod
  def type(self) -> PartType:
    raise NotImplementedError
```